dali/test/operators/passthrough_input.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TEST_OPERATORS_PASSTHROUGH_INPUT_H_
+#define DALI_TEST_OPERATORS_PASSTHROUGH_INPUT_H_
+
+#include <vector>
+
+#include "dali/pipeline/operator/builtin/input_operator.h"
+
+namespace dali {
+
+
+template<typename Backend>
+class PassthroughInput : public InputOperator<Backend> {
+  using OutBackend = std::conditional_t<
+          std::is_same_v<Backend, CPUBackend>,
+          CPUBackend /* CPUBackend */,
+          GPUBackend /* GPUBackend or MixedBackend */
+  >;
+
+ public:
+  explicit PassthroughInput(const OpSpec &spec) :
+          InputOperator<Backend>(spec),
+          cpu_input_(spec.GetArgument<bool>("cpu_input")) {
+    if constexpr (std::is_same_v<Backend, MixedBackend>) {
+      tp_ = std::make_unique<ThreadPool>(this->num_threads_, this->device_id_, false,
+                                         "PassthroughInput thread pool");
```

Comment:
why don't you use the thread pool from the workspace. Mixed operators should have one.

Reply:
At some point of the implementation I've got the error message:
```
Error when executing Mixed operator PassthroughInput encountered:
[dali/pipeline/workspace/workspace.h:393] Assert on "HasThreadPool()" failed: Workspace does not have a Thread Pool
```
Thus I figured, that Mixed operator do not have the thread pool.

The same situation is in the VideoDecoder operator, which is also a mixed one: https://github.com/NVIDIA/DALI/blob/51d6ff95c7380b5a8309c588eb03085ff10e6338/dali/operators/decoder/video/video_decoder_mixed.h#L52